src/tiled/stampbrush.cpp

```diff
@@ -250,9 +250,17 @@ void StampBrush::endCapture()
 
     // Intersect with the layer and translate to layer coordinates
     QRect captured = capturedArea();
+
     captured &= QRect(tileLayer->x(), tileLayer->y(),
                       tileLayer->width(), tileLayer->height());
 
+    //gets if the relative stagger should be the same as the base layer
+    int staggerIndexOffSet;
+    if (tileLayer->map()->staggerAxis() == Map::StaggerX)
+        staggerIndexOffSet = captured.x() % 2;
+    else
+        staggerIndexOffSet = captured.y() % 2;
```

Comment:
Please move this bit of code down to after `if (captured.isValid()) {`, since you don't need the variable yet here.